tools/scripts/Makefile

```diff
@@ -0,0 +1,265 @@
+$(EXEC): check-srcs $(subst LOCAL_PLATFORM,xilinx,$(OBJS))		\
+	$(BUILD_DIR)/app/src/lscript.ld
+	$(ECHO) [LD] $(shell ls $(OBJECTS_DIR)) '\n'
+	$(MUTE)$(CC) $(LDFLAGS) $(LIB_PATHS) 				\
+	$(OBJECTS_DIR)/*.o						\
+	$(LIBS) -o $(BUILD_DIR)/$(EXEC).elf
+
+# Copy the source files from the no-Os repo to the local project
+define copy_srcs
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(SRCS))		\
+	$(BUILD_DIR)/app/src/
+	$(MUTE)cp -r $(subst LOCAL_PLATFORM,$(1),$(INCS))		\
+	$(BUILD_DIR)/app/src/
+endef
+
+.DEFAULT_GOAL := all
+all: 	xilinx
+
+compile: $(EXEC)
+
+create-obj-dir:
+	$(MUTE)mkdir -p $(OBJECTS_DIR)
+	
+create-build-dir:
+	$(MUTE)mkdir -p $(BUILD_DIR)
+
+copy-srcs:
+	$(call copy_srcs,$(PLATFORM))
+
+check-srcs:
+ifeq (xilinx,$(strip $(PLATFORM)))
+# Replace the target sources (the default ones are the repository files) with
+# the local project sources
+	$(eval SRCS :=$(shell find $(BUILD_DIR)/app/src/ -name '*.c'))
+	$(eval EXTRA_INCS :=-I $(BUILD_DIR)/bsp/$(PROCESSOR_CORE)/include)
+	$(eval INC_PATHS :=-I $(BUILD_DIR)/app/src)
+endif
+
+clean:
+	$(ECHO) Cleaning build workspace '\n'
+	$(MUTE)rm -rf temp
+	$(MUTE)rm -rf $(BUILD_DIR)
+	@$(MAKE) -s xil-pre-cleanup
+
+eval-hardware:
+# If the hardware file is not specified, start searching for one
+# Check for .hdf files inside the project directory		
+ifndef $(HARDWARE)
+	$(eval HARDWARE = $(shell					\
+	if [ -z $(HARDWARE) ]; then					\
+		echo $(shell find $(PROJECT_DIR) -name "*.hdf") ;	\
+	fi;))
+	$(eval HARDWARE = $(shell					\
+	if [ -z $(HARDWARE) ]; then					\
+		echo $(shell find $(PROJECT_DIR) -name "*.sopcinfo") ;	\
+	else								\
+		echo $(shell find $(PROJECT_DIR) -name "*.hdf") ;	\
+	fi;))
+endif
+# Assign the platform based on the hardware file extension
+	$(ECHO) Evaluating hardware : $(notdir $(HARDWARE))\\n
+	$(eval PLATFORM:= $(shell \
+	if [ "$(findstring hdf,$(notdir $(HARDWARE)))" = "hdf" ]; then	\
+		echo xilinx;						\
+	elif [ "$(findstring sopcinfo,$(notdir $(HARDWARE)))" = "sopcinfo" ]; then\
+		echo altera;						\
+	else								\
+		echo none;						\
+	fi;))
+	@ if [ "$(PLATFORM)" = "none" ]; then				\
+		echo  \\e[32m[$(TIMESTAMP)]\\e[31m Platform not found\\e[39m;\
+		exit 1;							\
+	fi;
+	$(ECHO) "Found platform : \e[33m$(PLATFORM)\e[39m\\n";
+
+pre-build:
+	@$(MAKE) -s xil-pre-cleanup
+	@$(MAKE) -s create-build-dir
+	@$(MAKE) -s create-obj-dir
+
+prepare-project:
+	@ if [ "$(PLATFORM)" = "xilinx" ];then				\
+		$(MAKE) -s xil-prepare-project	;			\
+		$(MAKE) -s copy-srcs ;					\
+	fi;
+	@$(MAKE) -s compile
+
+xilinx:	xil-pre-cleanup							\
+	eval-hardware							\
+	pre-build	 						\
+	xil-prepare-project
+	$(call copy_srcs,xilinx)
+	@$(MAKE) -s compile
+	$(MUTE)rm -rf $(BUILD_DIR)/SDK.log
+	$(ECHO) Done \\n
+
+xil-extract-hdf-info:
+	$(MUTE)mkdir -p temp
+	$(MUTE)cp $(HARDWARE) temp
+	$(MUTE)rm -rf arch.txt
+# Extract the processor name from the hdf file
+	$(MUTE)xsct $(SCRIPTS_PATH)/read_hdf.tcl $(NULL)		\
+	temp/$(notdir $(HARDWARE))
```

Comment:
if breaking a new line here, maybe indent the 2nd, to make it more obvious to see they are the same